### ISSUE TITLE
"C_Cpp: Dim Inactive Regions" removes code folding arrows in inactive regions

### ISSUE THREAD

**Issue by svojvodic** [microsoft/vscode-cpptools#8613]
We have written the needed data into your clipboard because it was too large to send. Please paste.
Issue Type: <b>Bug</b>

- Open a C file with an inactive region, for example:

```
#ifdef inactive

int fun1() {
    return 1;
}

#else

int fun2() {
    return 2;
}

#endif
```

- Enable "C_Cpp: Dim Inactive Regions"
- **Expected:** A folding arrow appears for fun1, fun2, ifdef, else
- **Actual:** A folding arrow appears for fun2, ifdef, else, but not for fun1

![inactive](https://user-images.githubusercontent.com/4761803/147786611-b9608c72-020b-46d2-b2c4-ba3ed8c5d175.png)

Extension version: 1.7.1
VS Code version: Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:39:46.686Z)
OS version: Linux x64 5.4.0-91-generic
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz (12 x 2591)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|25.43GB (0.40GB free)|
|Process Argv|--unity-launch --crash-reporter-id e0b08cfb-692d-46c2-a2e8-1983e42a5725|
|Screen Reader|no|
|VM|100%|
|DESKTOP_SESSION|cinnamon|
|XDG_CURRENT_DESKTOP|X-Cinnamon|
|XDG_SESSION_DESKTOP|cinnamon|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30384385
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
vscop804:30404766
vscop341:30404997
vsrem710cf:30416617
vscexrecpromp3c:30407761

```

</details>

**Comment by sean-mcmanus** [comment]
The behavior is inherited from VS, since inactive regions information is from IntelliSense, which is disabled in the inactive region.

**Comment by svojvodic** [comment]
A thing to note: setting the "C_Cpp: Dim Inactive Regions" option to false enables the arrows. The region is still considered inactive by IntelliSense:
![dimming disabled](https://user-images.githubusercontent.com/4761803/148118643-9f90651f-d99e-47ca-a33f-36919d3ceaa6.png)

**Comment by sean-mcmanus** [comment]
I filed a VS feature request at https://developercommunity.visualstudio.com/t/Enable-outlining-regions-in-inactive-reg/1628686 . However, your observation that setting dimInactiveRegions to false causing the code folding to work implies that there might be a quick fix we can do, so I'll see if that's the case or not...

**Comment by sean-mcmanus** [comment]
The current behavior looks related to the fix for https://github.com/microsoft/vscode-cpptools/issues/5429 , i.e. the bug repros again when dimInactiveRegions is false.

I'll see what @Colengms has to say about it when he gets back...

**Comment by sean-mcmanus** [comment]
I'm guessing the current implementation was done so that inactive regions with `{` and `}` don't break the outlining of active regions. Theoretically we could add a setting to revert to the alternative behavior if outlining in inactive regions is more desirable than that issue.

**Comment by svojvodic** [comment]
I believe a safe improvement would be to allow folding blocks which are completely contained in a continuous region.
Making a complete solution which enables folding inside inactive regions and which is sensitive to issue described in #5429 would definitely be an interesting challenge.

**Comment by sean-mcmanus** [comment]
Yeah, someone on our team said they would take a look at that doing something similar to your suggestion.
